modules/auth/proc-adding-oauth2-proxy-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,83 @@
+[id='proc-adding-oauth2-proxy-as-an-authentication-provider_{context}']
+= Adding OAuth2 proxy as an authentication provider in {product}
+
+The {product} `core-plugin-api` package includes an OAuth2-proxy authentication provider
+that can authenticate users
+by using an link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2 proxy] in front of an {product} instance.
+This enables to reuse existing authentications within a cluster.
+In general, the OAuth2 proxy supports all OpenID Connect providers.
+For more details, see link:https://oauth2-proxy.github.io/oauth2-proxy/configuration/providers/[the supported provider list].
+
+.Procedure
+
+. Set up the OAuth2-proxy authentication provider in the {product} backend (`app-config.yaml`).
++
+Add the `oauth2Proxy` provider configuration to your `app-config.yaml` under the root `auth` configuration:
```

Comment:
We should align working with other auth providers. For example in Azure auth provider ther is 

>Edit your custom {product-short} config map, such as `app-config-rhdh`.

>In the `app-config-rhdh.yaml` content, add the `microsoft` provider configuration >under the root `auth` configuration, and enable the `microsoft` provider for sign-in: